examples/golang/object-upload.go

```diff
@@ -0,0 +1,78 @@
+package main
+
+import (
+	"fmt"
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/aws/endpoints"
+	"github.com/aws/aws-sdk-go/aws/session"
+	"github.com/aws/aws-sdk-go/service/s3/s3manager"
+	"os"
+)
+
+
+func main(){
+
+	if len(os.Args) != 3 {
+        exitErrorf("bucket and file name required\nUsage: %s bucket_name filename",
+            os.Args[0])
+    }
+
+    bucket := os.Args[1]
+    filename := os.Args[2]
+
+    file, err := os.Open(filename)
+    if err != nil {
+        exitErrorf("Unable to open file %q, %v", filename, err)
+    }
+
+    defer file.Close()
+
+	defaultResolver := endpoints.DefaultResolver()
+	s3CustResolverFn := func(service, region string, optFns ...func(*endpoints.Options)) (endpoints.ResolvedEndpoint, error) {
+		if service == "s3" {
+			return endpoints.ResolvedEndpoint{
+				URL:           "http://127.0.0.1:8000", //Variable substituing endpoint-url
+			}, nil
+		}
+
+		return defaultResolver.EndpointFor(service, region, optFns...)
+	}
+
+	sess := session.Must(session.NewSessionWithOptions(session.Options{
```

Comment:
this is working only because you have set the vstart credentials via `aws configure`
please add the credentials as parameters to the program so we don't rely on the aws ALI

Reply:
I just added the credentials as static parameters.